server/pulp/server/webservices/views/tasks.py

```diff
@@ -67,6 +67,32 @@ def get(self, request):
         serialized_task_statuses = [task_serializer(task) for task in raw_tasks]
         return generate_json_response_with_pulp_encoder(serialized_task_statuses)
 
+    @auth_required(authorization.DELETE)
+    def delete(self, request):
+        """
+        Delete the tasks with status as finished, error, timed-out
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+
+        :return: Status of delete operation or pulp Exception
+        :rtype:  String or pulp.Exception
+        """
+        task_state = request.GET.getlist('state')
+        if not task_state:
+            raise pulp_exceptions.PulpCodedForbiddenException()
+        complete_states = ("skipped", "error", "finished", "timed out")
+        for state in task_state:
+            if state not in complete_states:
+                raise pulp_exceptions.PulpCodedForbiddenException()
+            try:
+                TaskStatus.objects(state=state).delete()
+                result = "Tasks Deleted Successfully"
+            except:
```

Comment:
This could mask all kinds of other problems. What if there was an error connecting to the database, for example? Or what if the database experiences some internal error, like it timed out because there were too many records to delete? In general, you should almost never have a naked ``except`` statement in python.

I'd specifically catch the exception you expect if no resources are found. Although would mongoengine actually raise such an exception in this case?

Otherwise, let any other exception bubble up and get caught by the middleware. That should turn uncaught exceptions into 5xx responses.